certgen/certProcessor/src/main/java/org/incredible/certProcessor/CertificateFactory.java

```diff
@@ -60,11 +60,18 @@ public CertificateExtension createCertificate(CertModel certModel, Map<String, S
         /**
          * badge class object
          * **/
-
-        badgeClassBuilder.setName(certModel.getCertificateName()).setDescription(certModel.getCertificateDescription())
+        badgeClassBuilder.setDescription(certModel.getCertificateDescription())
                 .setId(properties.get(JsonKey.BADGE_URL)).setCriteria(certModel.getCriteria())
                 .setImage(certModel.getCertificateLogo()).
                 setIssuer(issuerBuilder.build());
+        //TODO for now badge name is set as course name, it should be certificate name
+        //after certificate QR validation the portal uses badge name to display the course name,but course name is in the training evidence(introduced in release-1.5.0)
+        //in evidence we are setting training name as courseName
+        if(StringUtils.isNotEmpty(certModel.getCourseName())){
```

Comment:
The training evidence is filled because 'courseName' is passed.
The assessment evidence is to be filled because 'quiz/assessName' is passed (future todo).
The certificate name (badge name) is mandatory and usually is 'completion certification', 'appreciation certificate'. Upon scan, there must be refined logic to pick and show relevant name. Until then, this workaround is in place.